dask-gateway-server/Dockerfile

```diff
@@ -1,45 +1,41 @@
-FROM python:3.9-slim-bullseye as dependencies
-LABEL MAINTAINER="Jim Crist-Harif"
-
+# This Dockerfile and image, ghcr.io/dask/dask-gateway-server, is used by the
+# dask-gateway Helm chart, by the api pod and the controller pod.
+#
+# The pods are started with different commands:
+#
+# - api pod command:        dask-gateway-server ...
+# - controller pod command: dask-gateway-server kube-controller ...
+#
+FROM python:3.10-slim-bullseye
+
+# Install tini and upgrade linux packages are updated to patch known
+# vulnerabilities.
 RUN apt-get update \
-    && apt-get install -y tini \
-    && rm -rf /var/lib/apt/lists/*
-
+ && apt-get upgrade -y \
```

Comment:
A full upgrade seems heavy handed - isn't equivalent to starting with a newer distribution?

Reply:
Ah no this is quite trivial, it will only do small upgrades for known vulnerabilities in my experience from doing it in jupyterhub/zero-to-jupyterhub-k8s. The official images on dockerhub are slow to update, so this can be helpful for such situations.

Reply:
This is what changed when I did built this image:

```Dockerfile
FROM python:3.10-slim-bullseye

RUN apt-get update \
 && apt-get upgrade -y
```

```
Step 1/2 : FROM python:3.10-slim-bullseye
 ---> ea290afaeb03
Step 2/2 : RUN apt-get update  && apt-get upgrade -y
 ---> Running in 99bfddeb5dc1
Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:2 http://security.debian.org/debian-security bullseye-security InRelease [44.1 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [39.4 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8182 kB]
Get:5 http://security.debian.org/debian-security bullseye-security/main amd64 Packages [126 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [2596 B]
Fetched 8510 kB in 2s (3930 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
The following packages will be upgraded:
  gzip liblzma5
2 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Need to get 300 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://security.debian.org/debian-security bullseye-security/main amd64 gzip amd64 1.10-4+deb11u1 [132 kB]
Get:2 http://security.debian.org/debian-security bullseye-security/main amd64 liblzma5 amd64 5.2.5-2.1~deb11u1 [168 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 300 kB in 1s (469 kB/s)
(Reading database ... 7023 files and directories currently installed.)
Preparing to unpack .../gzip_1.10-4+deb11u1_amd64.deb ...
Unpacking gzip (1.10-4+deb11u1) over (1.10-4) ...
Setting up gzip (1.10-4+deb11u1) ...
(Reading database ... 7023 files and directories currently installed.)
Preparing to unpack .../liblzma5_5.2.5-2.1~deb11u1_amd64.deb ...
Unpacking liblzma5:amd64 (5.2.5-2.1~deb11u1) over (5.2.5-2) ...
Setting up liblzma5:amd64 (5.2.5-2.1~deb11u1) ...
Processing triggers for libc-bin (2.31-13+deb11u3) ...
Removing intermediate container 99bfddeb5dc1
```